packages/react/src/components/UIShell/SideNav.js

```diff
@@ -119,13 +119,22 @@ const SideNav = React.forwardRef(function SideNav(props, ref) {
     eventHandlers.onMouseLeave = () => handleToggle(false, false);
   }
 
+  useEffect(() => {
+    console.log(isFixedNav);
+    if (expanded && !isFixedNav) {
+      document.getElementById(`${prefix}--side-nav__navigation`).focus();
```

Comment:
I think this should work? if not you can declare your own ref and `useMergedRefs` (see screenshot from [Dialog](https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/Dialog/index.js#L22)) to pass onto the div instead of `ref={ref}` in line 138. This should also eliminate the need for setting an `id` in the `div` element
```suggestion
{
      ref.current?.focus();
}
```
<img width="741" alt="image" src="https://user-images.githubusercontent.com/40550942/234648572-e127e804-b963-43d3-ab03-136ef03da70a.png">